### PR TITLE
Include Nuage OpenShift Monitors in Loadbalancer Config file

### PR DESCRIPTION
**Pull request by vareti**
PR has changes to 
1. Include `nuage_node` role during additional node config playbook
2. Include `kube_proxy_and_dns` role to start `kube-proxy`  in `nuage_node` when Nuage SDN is used
3. Use `nuage_openshift_monitor_rest_server_port` instead of `nuage_mon_rest_server_port` in loadbalancer  config file. This also fixes the issue of [nuage monitor](https://github.com/nuagenetworks/nuage-kubernetes/tree/master/nuagekubemon) not being included  in haproxy config file
4. Stop openvswitch service when Nuage VRS pod is terminated
5. Add new env variable `NUAGE_K8S_POD_NETWORK_CIDR` to Nuage VRS pod.
6. Use `openshift_portal_net` and `osm_cluster_network_cidr` for Nuage service and pod network CIDRs instead of using separate variables in inventory file 
7. Changes to include `openshift_facts` in `nuage_node` role
8. Redeploy etcd certificates for Nuage whenever ansible is re-run

### CODE REVIEW

**Comment by vrutkovs** [comment]
Please create a PR against `release-3.11` branch instead

**Comment by vareti** [comment]
@vrutkovs Thanks for the review. Updated the comment description to describe the PR

File: roles/nuage_master/vars/main.yaml [hunk 1]
```diff
@@ -35,8 +35,10 @@ api_server_url: "{{ hostvars[groups.oo_first_master.0].openshift.master.api_url
 nuage_vport_mtu: "{{ nuage_interface_mtu | default('1460') }}"
 master_host_type: "{{ master_base_host_type | default('is_rhel_server') }}"
 slave_host_type: "{{ slave_base_host_type | default('is_rhel_server') }}"
-svc_ipv4_subnet: "{{ nuage_svc_ipv4_subnet | default('172.30.0.0') }}"
-svc_ipv4_subnet_mask: "{{ nuage_svc_ipv4_subnet_mask | default('16') }}"
+svc_ipv4_subnet: "{{ openshift_portal_net.split('/')[0] | default('172.30.0.0') }}"
+svc_ipv4_subnet_mask: "{{ openshift_portal_net.split('/')[1] | default('16') }}"
+pod_network_subnet : "{{ osm_cluster_network_cidr.split('/')[0] | default('70.70.0.0') }}"
```
Comment (vrutkovs) [inline roles/nuage_master/vars/main.yaml]:
```
  40:19     error    too many spaces before colon  (colons)
```
```suggestion
pod_network_subnet: "{{ osm_cluster_network_cidr.split('/')[0] | default('70.70.0.0') }}"
```

**Comment by vareti** [comment]
@vrutkovs We tested the PR locally and we were able to install OpenShift-3.11 with Nuage SDN solution. Could you please review the PR when you get a chance?

**Comment by pdellaert** [comment]
Thanks, @sdodson 
I'll have @vareti comment on the failed tests, but looking at those tests, i think this is because Nuage is a proprietary solution and we do not have our images available publicly. You also need a Nuage management and control plane before any CNI based tests can be successful.

**Comment by sdodson** [comment]
The CI failures are related to a Centos repo problem and unrelated to Nuage entirely. Hopefully they'll be resolved shortly.

**Comment by pdellaert** [comment]
@sdodson , sorry for the extra changes, while running our full QA Regression cycle, we did find one more issue. That has been fixed now.

We've run through several testing cycles again and everything looks good now. Could you review again?

**Comment by pdellaert** [comment]
@vareti can you please urgently address this?

**Comment by vareti** [comment]
@vrutkovs @sdodson Could we rerun the failing tests and get the PR to merge

**Comment by pdellaert** [comment]
@sdodson, is there anything we can do about this loop that seems to be occurring?

**Pull request merged**